x-pack/plugins/data_enhanced/server/search/session/check_running_sessions.ts

```diff
@@ -93,8 +93,14 @@ async function updateSessionStatus(
   // And only then derive the session's status
   const sessionStatus = getSessionStatus(session.attributes);
   if (sessionStatus !== session.attributes.status) {
+    const now = new Date().toISOString();
     session.attributes.status = sessionStatus;
-    session.attributes.touched = new Date().toISOString();
+    session.attributes.touched = now;
+    if (sessionStatus === SearchSessionStatus.COMPLETE) {
+      session.attributes.completed = now;
+    } else if (session.attributes.completed) {
```

Comment:
Could you explain this case please?

Reply:
This branch is for the case when we transition from `Completed` to something else:
`Completed` -> (new search was added) -> `In-progress` (mostly for this case)
`Completed` -> `Expired` (for this case I think it would be find to leave `completed` time, but for consistency with previous case I am also removing it)

Reply:
In two words: 

I wanted `completed` to exist only when session is in `COMPLETE` state